app/assets/javascripts/helpers/profile_photo_uploader.es6

```diff
@@ -0,0 +1,250 @@
+     * @param text - The text to display.
+     */
+    this.showMessage = (type, text) => { return app.flashMessages ? app.flashMessages[type](text) : alert(text); };
+
+    this.initFineUploader();
+  }
+
+  /**
+   * Initializes the fine uploader component
+   */
+  initFineUploader() {
+    this.fineUploader = new qq.FineUploaderBasic({
+      element: this.fileInput,
+      validation: {
+        allowedExtensions: ["jpg", "jpeg", "png"]
+      },
+      request: {
+        endpoint: Routes.photos(),
+        params: {
+          /* eslint-disable camelcase */
+          authenticity_token: $("meta[name='csrf-token']").attr("content"),
+          /* eslint-enable camelcase */
+          photo: {"pending": true, "aspect_ids": "all", "set_profile_photo": true}
+        }
+      },
+      button: this.fileInput,
+      autoUpload: false,
+
+      messages: {
+        typeError: Diaspora.I18n.t("photo_uploader.invalid_ext"),
+        sizeError: Diaspora.I18n.t("photo_uploader.size_error"),
+        emptyError: Diaspora.I18n.t("photo_uploader.empty")
+      },
+
+      callbacks: {
+        onProgress:
+          (id, fileName, loaded, total) => (this.info.innerText = `${fileName} ${Math.round(loaded / total * 100)}%`),
+        onSubmit: (id, name) => this.onPictureSelected(id, name),
+        onComplete: (id, fileName, responseJSON) => this.onUploadCompleted(id, fileName, responseJSON),
+        onError: (id, name, errorReason) => this.showMessage("error", errorReason)
+      },
+
+      text: {
+        fileInputTitle: ""
+      },
+
+      scaling: {
+        sendOriginal: false,
+
+        sizes: [
+          {maxSize: 1600}
+        ]
+      }
+    });
+  }
+
+  /**
+   * Called when a picture from user's device has been selected.
+   * @param {number} id - The current file's id.
+   * @param {string} name - The current file's name.
+   */
+  onPictureSelected(id, name) {
+    this.setLoading(true);
+    this.fileName = name;
+    const file = this.fileInput.querySelector("input").files[0];
+
+    // ensure browser's file reader support
+    if (FileReader && file) {
+      try {
+        const fileReader = new FileReader();
+        fileReader.onload = () => this.initCropper(fileReader.result);
+        fileReader.readAsDataURL(file);
+      } catch (e) {
+        this.setLoading(false);
+      }
+    } else {
+      this.setLoading(false);
+    }
+  }
+
+  /**
+   * Initializes the cropper and all controls.
+   * @param {object|string} imageData - The base64 image data
+   */
+  initCropper(imageData) {
+    // cache the current picture source if the user cancels
+    this.previousPicture = this.picture.getAttribute("src");
+
+    this.mimeType = imageData.split(";base64")[0].substring(5);
+
+    this.picture.onload = () => {
+      // set the preferred size style of the cropper based on picture orientation
+      this.picture.setAttribute("style", (this.picture.naturalHeight > this.picture.naturalWidth ?
```

Comment:
I wouldn't mind having this written out a bit more verbose to avoid line breaks inside ternaries.

Reply:
I just pushed an update and improved this.